messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -392,6 +396,11 @@ private void processBatch(Segment segment, BlockingStream<TrackedEventMessage<?>
             unitOfWork.attachTransaction(transactionManager);
             unitOfWork.resources().put(segmentIdResourceKey, segment.getSegmentId());
             unitOfWork.resources().put(lastTokenResourceKey, finalLastToken);
+            logger.trace("Processor {} processing batch {}. Segment id {}. Thread {}.",
```

Comment:
There's quite a bit of information-gathering in this `logger.trace` statement. I would suggest adding a logger.isTraceEnabled() around it.